homeassistant/components/plex/config_flow.py

```diff
@@ -0,0 +1,158 @@
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialize the Plex flow."""
+        self.current_login = {}
+        self.available_servers = None
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=USER_SCHEMA, errors={}
+            )
+
+        return await self.async_step_server_validate(user_input)
+
+    async def async_step_server_validate(self, server_config):
+        """Validate a provided configuration."""
+        errors = {}
+        self.current_login = server_config
+
+        plex_server = PlexServer(server_config)
+        try:
+            await self.hass.async_add_executor_job(plex_server.connect)
+        except NoServersFound:
+            errors["base"] = "no_servers"
+        except ServerNotSpecified as available_servers:
+            self.available_servers = available_servers.args[0]
+            return await self.async_step_select_server()
+        except (plexapi.exceptions.BadRequest, plexapi.exceptions.Unauthorized):
+            _LOGGER.error("Invalid credentials provided, config not created")
+            errors["base"] = "faulty_credentials"
+        except (plexapi.exceptions.NotFound, requests.exceptions.ConnectionError):
+            _LOGGER.error(
+                "Plex server could not be reached: %s", server_config[CONF_URL]
+            )
+            errors["base"] = "not_found"
+        except Exception as error:  # pylint: disable=broad-except
+            _LOGGER.error("Unknown error connecting to Plex server: %s", error)
+            return self.async_abort(reason="unknown")
+
+        if errors:
+            return self.async_show_form(
+                step_id="user", data_schema=USER_SCHEMA, errors=errors
+            )
+
+        server_id = plex_server.machine_identifier
+
+        for entry in self._async_current_entries():
+            if entry.data[CONF_SERVER_IDENTIFIER] == server_id:
+                return self.async_abort(reason="already_configured")
+
+        url = plex_server.url_in_use
+        token = server_config.get(CONF_TOKEN)
+
+        entry_config = {CONF_URL: url}
+        if token:
+            entry_config[CONF_TOKEN] = token
+        if url.startswith("https"):
+            entry_config[CONF_VERIFY_SSL] = server_config.get(
+                CONF_VERIFY_SSL, DEFAULT_VERIFY_SSL
+            )
+
+        _LOGGER.debug("Valid config created for %s", plex_server.friendly_name)
+
+        return self.async_create_entry(
+            title=plex_server.friendly_name,
+            data={
+                CONF_SERVER: plex_server.friendly_name,
+                CONF_SERVER_IDENTIFIER: server_id,
+                PLEX_SERVER_CONFIG: entry_config,
+            },
+        )
+
+    async def async_step_select_server(self, user_input=None):
```

Comment:
Look at hue config flow on how to make the select step a bit more readable

Reply:
Hopefully addressed, please let me know if I should add more.

(It helped me uncover a new edge case, thanks for the suggestion!)

Reply:
Looks much better 👍